obspy/clients/fdsn/mass_downloader/download_helpers.py

```diff
@@ -696,7 +706,17 @@ def star_download_station(args):
                 starttime = min([_i.starttime for _i in c_info])
                 endtime = max([_i.endtime for _i in c_info])
                 if starttime > times[0] or endtime < times[1]:
-                    still_missing[c_id] = times
+                    # Cope with case that not full day of station info missing
+                    if starttime < times[1]:
+                        still_missing[c_id] = (times[0], starttime)
+                        station.have_station_information[c_id] = (starttime,
+                                                                  times[1])
+                    elif endtime > times[0]:
+                        still_missing[c_id] = (endtime, times[1])
+                        station.have_station_information[c_id] = (times[0],
+                                                                  endtime)
+                    else:
+                        still_missing[c_id] = times
```

Comment:
This was my simple way - but I think an attribute could be better. Can't think of a simple way to cope with beginning and end missing for a single channel - that would be the case where there is station info for the middle of a day or data, but not either end... Would be an odd case, but for very short-term deployments it might happen?